src/content/drafts/sourcecode-to-tokens-AST.md

```diff
@@ -0,0 +1,97 @@
+
+In this tutorial, first, you'll learn on a high-level how your source code is broken down into multiple tokens and is then converted to an AST. Next, you'll learn how [ESLint](https://eslint.org/docs/latest/) leverages the capability of AST and helps you build your own custom rules. This tutorial is intended for those who use ESLint and wants to understand more about how the tool works. It assumes that you have basic familiarity with:
+- Computer programming
+- Javascript coding
+- ESLint tool
+  
+By the end of this tutorial, you’ll be able to:
+- Understand how the source code is converted to AST.
+- Generate AST for your source program using AST explorer.
+- Understand how to traverse the AST nodes to write custom rules in ESLint.
+
+## Step 1: Lexical analysis
+[Lexical analysis](https://en.wikipedia.org/wiki/Lexical_analysis) commonly known as tokenization is the process of converting your source code into a sequence of tokens. A token can be any valid entity in a programming language such as an identifier, keyword, operator, function call, string, or separator. Tokens do not understand how things fit together instead focus on different components of a file. Consider it analogous to breaking a sentence into different words and identifying the words such as nouns, verbs, numbers, adjectives, and punctuation. At this point, you don’t understand what’s part of a sentence or how these words fit together to make a sentence. The below picture depicts how a simple `if statement` is converted into multiple tokens.
+
+![If statement converted to tokens](/src/assets/images/blog/2022/lexical-analysis.png "source code to tokens")
+
+## Step 2: Syntax analysis
+In this step, the list of tokens is converted into an AST - tree-like data structure that represents the actual structure of the code.
+
+![AST](/src/assets/images/blog/2022/AST-visual.png "AST")
+The AST visual shown above is generated using [JointJS](https://resources.jointjs.com/demos/javascript-ast).
+
+ You can consider this step analogous to finding the relationship between nouns, verbs, and adjectives in a sentence and identifying how they conform to the language’s grammar.  Continuing with the same example, in the previous step, there were three individual tokens for x, >, and 18 but after syntax analysis, it can be inferred that x>18 is a condition for an `if statement`. 
+
+![A list of tokens converted to AST](/src/assets/images/blog/2022/syntax-analysis.png "tokens to AST")
+
+In the above example,[ AST explorer](https://astexplorer.net/#/gist/052d25ec2db5e45442e474ffc8dd0f43/latest) is used for visualizing the AST for a simple JavaScript code snippet. AST explorer is a freely available online tool that can be used to generate AST for different programming languages.
+
+:information_source:**Tip**: If you ever get lost in the AST, you can hover over different sections of your source code on the right and the tool then highlights the relevant node in the AST on the left.
```

Comment:
`:information_source:` Just wanted to confirm if we were trying to transform this into different icon or something else? But it renders as such in preview.

Reply:
:information_source: is a markdown emoji. It will be rendered as ℹ️  I thought instead of writing **Note**  I could use this emoji. It does not render in the VSCode preview though.

Reply:
I did not use VSCode preview. I used the netlify build preview and it didn't show up. 
https://deploy-preview-324--new-eslint.netlify.app/content/drafts/sourcecode-to-tokens-ast/

Reply:
Okay! Can I change it to just Note?

Reply:
I see in other parts of the documentation we have used `**Note:**` may be we can use the same

Reply:
I have changed the emoji to **Note:**